lib/Config/Migrator.php

```diff
@@ -427,11 +428,12 @@ public function statisticsClicks() {
       'queue_id int(11) unsigned NOT NULL,',
       'link_id int(11) unsigned NOT NULL,',
       'user_agent_id int(11) unsigned NULL,',
+      'user_agent_type varchar(50) NULL,',
```

Comment:
Would it be better to use an enum column here? This table was fixed length before this change. Now it isn't. I know there is a difference in MySQL. I don't remember exactly what the connotations are.

Also, the user agent strings are stored in the user agents table. Why is the type stored here? Why don't we store the type in the user agents table?

Reply:
You are right with the column type. I would change the type on the column on tinyint with constants in the code.

> Also, the user agent strings are stored in the user agents table. Why is the type stored here? Why don't we store the type in the user agents table?

This was recommended in the ticket description, and it makes sense to me when we want to have a good performance on bigger sites. But we can't be sure that the join would have been a problem.